### ISSUE TITLE
file_put_contents(): Exclusive locks are not supported

### ISSUE THREAD

**Issue by Jam0r** [Mibew/mibew#109]
When running on my host (cloud hosting) I get the following error

Warning: file_put_contents(): Exclusive locks are not supported for this stream in /var/sites/s/steve-morris1.co.uk/public_html/mibew/vendor/tedivm/stash/src/Stash/Driver/FileSystem.php on line 245

On all pages in the admin section and on the chat window.

System information

You are using:
2.0.0-beta.2

Latest version:
2.0.0-beta.2

Installed localizations:
en 

Environment:
PHP 5.5.12 PDO/1.0.4dev pdo_mysql/1.0.2 gd

**Comment by JustBlackBird** [comment]
The problem you are talking about is caused by cache implementation used in Mibew. The cache uses files as data storage but there are no exclusive locks on NFS that's why you see the error.

I think we could provide an option to choose cache storage in Mibew. At the same time one should understand that there are many tasks with higher priority tasks now.

**Comment by Jam0r** [comment]
Thanks for the reply.

It certainly wasn't a complaint, just a report in case it hadn't been reported it before.

Is there anything that I could possibly do in the mean time?

**Comment by JustBlackBird** [comment]
I can only advise you to patch the code. But you should apply the patch any time you update Mibew.

The simplest way is comment cache initialization out. This way will definitely decrease the performance. Take a look at [lines 45-48](https://github.com/Mibew/mibew/blob/master/src/mibew/index.php#L45-L48) in of the index.php file. Just comment them out and the cache will be disabled.

Another way is to replace file storage with something else. I describe how  to use Memcached storage, but you can use any other storage that is supported by https://github.com/tedious/Stash. You should replace [lines 45-46](https://github.com/Mibew/mibew/blob/master/src/mibew/index.php#L45-L46) with the following lines:
```php
$cache_driver = new Stash\Driver\Memcache();
$cache_driver->setOptions(array('servers' => array('127.0.0.1', '11211')));
```
Do not forget to replace `127.0.0.1` and `11211` with your own values.

**Comment by JustBlackBird** [comment]
I've added support of Memcached to the Core within 22aaedd3be015bd0e03311deb2a71e0b94d8a72c .  Also you can turn the caching off though I don't recommend you to do so. See the [config file](https://github.com/Mibew/mibew/blob/master/src/mibew/configs/default_config.yml) for details.

The changes will be available in 2.0.0-beta.4. I close the issue because the original problem cannot be solved in a simple way and there is a work around for it now.
